uniffi_bindgen/src/lib.rs

```diff
@@ -252,27 +253,37 @@ pub fn run_tests<P: AsRef<Path>>(
     Ok(())
 }
 
-fn get_config(component: &ComponentInterface) -> Result<Config> {
+fn get_config(
+    component: &ComponentInterface,
+    idl_file: &Path,
+    config_file_override: Option<&Path>,
+) -> Result<Config> {
     let default_config: Config = component.into();
 
-    let pkg_path: PathBuf = env::var("CARGO_MANIFEST_DIR")
-        .expect("Missing $CARGO_MANIFEST_DIR, cannot load config file")
-        .into();
-
-    let config_file = pkg_path.join("uniffi.toml").canonicalize();
-
-    let config_file = match config_file {
-        Ok(f) => f,
-        Err(_) => return Ok(default_config),
+    let config_file: Option<PathBuf> = match config_file_override {
+        Some(cfg) => Some(PathBuf::from(cfg)),
+        None => {
+            let path_guess = idl_file
+                .parent()
+                .ok_or_else(|| anyhow!("no parent!"))?
+                .join("uniffi.toml");
+            match path_guess.canonicalize() {
+                Ok(f) => Some(f),
+                Err(_) => None,
+            }
```

Comment:
This assumes that the config file is only ever going to be `../uniffi.toml` relative to the `idl_file`.